# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1498590658 14400
#      Tue Jun 27 15:10:58 2017 -0400
# Node ID 3b68aeeabc40c2611abb2587976f75b4b06a96dc
# Parent  750e42a35adc243dd892fee4545cbebfc4543f31
rename some build scripts for consistency and move them to where they are used

* libgui/mk-default-qt-settings.in.sh: Rename from
build-aux/mk-default-qt-settings.in.sh.
* libinterp/mk-doc.pl: Rename from libinterp/gendoc.pl.
* libinterp/operators/mk-ops.sh: Rename from libinterp/mkops.
* scripts/mk-doc.pl: Rename from scripts/mkdoc.pl.
* libinterp/mk-builtins.pl: Rename from build-aux/mk-builtins.pl.
* liboctave/external/mk-f77-def.in.sh: Rename from
build-aux/mk-f77-def.in.sh.
* libinterp/corefcn/mk-mxarray-h.in.sh: Rename from
build-aux/mk-mxarray-h.in.sh.
* libinterp/mk-version-h.in.sh: Rename from
build-aux/mk-version-h.in.sh.
* libinterp/corefcn/genprops.awk: Rename from libinterp/genprops.awk.
* libinterp/corefcn/mk-errno-list.sh: Rename from
libinterp/mk-errno-list.
* libinterp/mk-pkg-add.sh: Rename from libinterp/mk-pkg-add.
* scripts/mk-pkg-add.sh: Rename from scripts/mk-pkg-add.

* Makefile.am, build-aux/module.mk, configure.ac, libgui/module.mk,
libinterp/corefcn/module.mk, libinterp/dldfcn/config-module.awk,
libinterp/module.mk, libinterp/operators/module.mk,
liboctave/external/module.mk, liboctave/external/ranlib/module.mk,
scripts/module.mk: Update.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -93,16 +93,18 @@ octetc_DATA =
 octlocale_DATA =
 
 DOC_TARGETS =
 
 BUILT_DISTFILES =
 BUILT_NODISTFILES =
 EXTRA_DIST =
 
+GEN_CONFIG_SHELL =
+
 BUILT_DISTFILES += \
   AUTHORS \
   BUGS \
   ChangeLog \
   HG-ID \
   INSTALL.OCTAVE
 
 EXTRA_DIST += \
diff --git a/build-aux/module.mk b/build-aux/module.mk
--- a/build-aux/module.mk
+++ b/build-aux/module.mk
@@ -1,36 +1,27 @@
 EXTRA_DIST += \
   %reldir%/OctJavaQry.class \
   %reldir%/OctJavaQry.java \
   %reldir%/changelog.tmpl \
   %reldir%/check-subst-vars.in.sh \
   %reldir%/find-defun-files.sh \
   %reldir%/find-files-with-tests.sh \
-  %reldir%/mk-builtins.pl \
-  %reldir%/mk-default-qt-settings.in.sh \
-  %reldir%/mk-f77-def.in.sh \
   %reldir%/mk-hg-id.sh \
-  %reldir%/mk-mxarray-h.in.sh \
   %reldir%/mk-octave-config-h.sh \
   %reldir%/mk-opts.pl \
-  %reldir%/mk-version-h.in.sh \
   %reldir%/move-if-change \
   %reldir%/stl_algo.h-fixed \
   %reldir%/subst-config-vals.in.sh \
   %reldir%/subst-cross-config-vals.in.sh \
   %reldir%/subst-default-vals.in.sh \
   %reldir%/subst-script-vals.in.sh \
   %reldir%/update-bug-status.sh
 
-GEN_CONFIG_SHELL = \
-  %reldir%/mk-default-qt-settings.sh \
-  %reldir%/mk-f77-def.sh \
-  %reldir%/mk-mxarray-h.sh \
-  %reldir%/mk-version-h.sh \
+GEN_CONFIG_SHELL += \
   %reldir%/subst-config-vals.sh \
   %reldir%/subst-cross-config-vals.sh \
   %reldir%/subst-default-vals.sh \
   %reldir%/subst-script-vals.sh
 
 $(GEN_CONFIG_SHELL) : %.sh : %.in.sh config.status
 	$(AM_V_GEN)$(SHELL) config.status $@-tmp $@
 
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -3151,20 +3151,20 @@ dnl We use a .in.h file for oct-conf-pos
 dnl the build tree and so that we don't have to add the -I${top_srcdir}
 dnl to any CPPFLAGS variables.
 
 AC_CONFIG_FILES([oct-conf-post.h-tmp:oct-conf-post.in.h])
 AC_CONFIG_COMMANDS([oct-conf-post.h],
   [$SHELL $srcdir/build-aux/move-if-change oct-conf-post.h-tmp oct-conf-post.h])
 
 OCTAVE_CONFIG_MOVE_IF_CHANGE_FILES([
-  build-aux/mk-default-qt-settings.sh
-  build-aux/mk-f77-def.sh
-  build-aux/mk-mxarray-h.sh
-  build-aux/mk-version-h.sh
+  libgui/mk-default-qt-settings.sh
+  liboctave/external/mk-f77-def.sh
+  libinterp/corefcn/mk-mxarray-h.sh
+  libinterp/mk-version-h.sh
   build-aux/subst-config-vals.sh
   build-aux/subst-cross-config-vals.sh
   build-aux/subst-default-vals.sh
   build-aux/subst-script-vals.sh])
 
 AC_OUTPUT
 
 ### Print a summary so that important information isn't missed.
diff --git a/build-aux/mk-default-qt-settings.in.sh b/libgui/mk-default-qt-settings.in.sh
rename from build-aux/mk-default-qt-settings.in.sh
rename to libgui/mk-default-qt-settings.in.sh
diff --git a/libgui/module.mk b/libgui/module.mk
--- a/libgui/module.mk
+++ b/libgui/module.mk
@@ -74,18 +74,18 @@ nodist_%canon_reldir%_liboctgui_la_SOURC
   -bindir $(bindir) \
   $(LIBOCTGUI_LINK_OPTS) \
   $(WARN_LDFLAGS)
 
 octetc_DATA += %reldir%/default-qt-settings
 
 octlocale_DATA += $(LOCALES)
 
-%reldir%/default-qt-settings: %reldir%/default-qt-settings.in build-aux/mk-default-qt-settings.sh | %reldir%/$(octave_dirstamp)
-	$(AM_V_GEN)$(call simple-filter-rule,build-aux/mk-default-qt-settings.sh)
+%reldir%/default-qt-settings: %reldir%/default-qt-settings.in %reldir%/mk-default-qt-settings.sh | %reldir%/$(octave_dirstamp)
+	$(AM_V_GEN)$(call simple-filter-rule,%reldir%/mk-default-qt-settings.sh)
 
 DIRSTAMP_FILES += \
   %reldir%/$(octave_dirstamp)
 
 define moc-command
   rm -f $@-t $@ && \
   ( echo "#if defined (HAVE_CONFIG_H)"; \
     echo '#  include "config.h"'; \
@@ -125,17 +125,18 @@ am__v_lrelease_1 =
 	$(AM_V_GEN)$(LRELEASE) $(LRELEASEFLAGS) $(AM_V_lrelease) -qm $@ $<
 
 DIRSTAMP_FILES += \
   %reldir%/languages/$(octave_dirstamp)
 
 %canon_reldir%_EXTRA_DIST += \
   $(TRANSLATIONS) \
   %reldir%/default-qt-settings.in \
-  %reldir%/liboctgui-build-info.in.cc
+  %reldir%/liboctgui-build-info.in.cc \
+  %reldir%/mk-default-qt-settings.in.sh
 
 EXTRA_DIST += $(%canon_reldir%_EXTRA_DIST)
 
 %canon_reldir%_CLEANFILES += \
   %reldir%/liboctgui-build-info.cc
 
 %canon_reldir%_DISTCLEANFILES += \
   %reldir%/default-qt-settings \
@@ -151,9 +152,13 @@ libgui-clean:
 libgui-distclean: libgui-clean
 	rm -f $(%canon_reldir%_DISTCLEANFILES)
 
 libgui-maintainer-clean: libgui-distclean
 	rm -f $(%canon_reldir%_MAINTAINERCLEANFILES)
 
 %reldir%/liboctgui-build-info.cc: %reldir%/liboctgui-build-info.in.cc HG-ID | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)$(build-info-commands)
+
+GEN_CONFIG_SHELL += \
+  %reldir%/mk-default-qt-settings.sh
+
 endif
diff --git a/libinterp/genprops.awk b/libinterp/corefcn/genprops.awk
rename from libinterp/genprops.awk
rename to libinterp/corefcn/genprops.awk
diff --git a/libinterp/mk-errno-list b/libinterp/corefcn/mk-errno-list.sh
rename from libinterp/mk-errno-list
rename to libinterp/corefcn/mk-errno-list.sh
diff --git a/build-aux/mk-mxarray-h.in.sh b/libinterp/corefcn/mk-mxarray-h.in.sh
rename from build-aux/mk-mxarray-h.in.sh
rename to libinterp/corefcn/mk-mxarray-h.in.sh
diff --git a/libinterp/corefcn/module.mk b/libinterp/corefcn/module.mk
--- a/libinterp/corefcn/module.mk
+++ b/libinterp/corefcn/module.mk
@@ -247,39 +247,39 @@ COREFCN_SRC = \
   %reldir%/zfstream.cc \
   $(NOINSTALL_COREFCN_INC)
 
 ## Special rules for sources which must be built before rest of compilation.
 
 %reldir%/defaults.h: %reldir%/defaults.in.h build-aux/subst-default-vals.sh | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)$(call simple-filter-rule,build-aux/subst-default-vals.sh)
 
-%reldir%/graphics.h: %reldir%/graphics.in.h libinterp/genprops.awk | %reldir%/$(octave_dirstamp)
+%reldir%/graphics.h: %reldir%/graphics.in.h %reldir%/genprops.awk | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
-	$(AWK) -f $(srcdir)/libinterp/genprops.awk $< > $@-t && \
+	$(AWK) -f $(srcdir)/%reldir%/genprops.awk $< > $@-t && \
 	mv $@-t $@
 
-%reldir%/graphics-props.cc: %reldir%/graphics.in.h libinterp/genprops.awk | %reldir%/$(octave_dirstamp)
+%reldir%/graphics-props.cc: %reldir%/graphics.in.h %reldir%/genprops.awk | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
-	$(AWK) -v emit_graphics_props=1 -f $(srcdir)/libinterp/genprops.awk $< > $@-t && \
+	$(AWK) -v emit_graphics_props=1 -f $(srcdir)/%reldir%/genprops.awk $< > $@-t && \
 	mv $@-t $@
 
-%reldir%/oct-errno.cc: %reldir%/oct-errno.in.cc | %reldir%/$(octave_dirstamp)
+%reldir%/oct-errno.cc: %reldir%/oct-errno.in.cc %reldir%/mk-errno-list.sh | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
 	if test -n "$(PERL)"; then \
-	  $(SHELL) $(srcdir)/libinterp/mk-errno-list --perl "$(PERL)" < $< > $@-t; \
+	  $(SHELL) $(srcdir)/%reldir%/mk-errno-list.sh --perl "$(PERL)" < $< > $@-t; \
 	elif test -n "$(PYTHON)"; then \
-	  $(SHELL) $(srcdir)/libinterp/mk-errno-list --python "$(PYTHON)" < $< > $@-t; \
+	  $(SHELL) $(srcdir)/%reldir%/mk-errno-list.sh --python "$(PYTHON)" < $< > $@-t; \
 	else \
 	  $(SED) '/@SYSDEP_ERRNO_LIST@/D' $< > $@-t; \
 	fi && \
 	mv $@-t $@
 
-%reldir%/mxarray.h: %reldir%/mxarray.in.h build-aux/mk-mxarray-h.sh | %reldir%/$(octave_dirstamp)
-	$(AM_V_GEN)$(call simple-filter-rule,build-aux/mk-mxarray-h.sh)
+%reldir%/mxarray.h: %reldir%/mxarray.in.h %reldir%/mk-mxarray-h.sh | %reldir%/$(octave_dirstamp)
+	$(AM_V_GEN)$(call simple-filter-rule,%reldir%/mk-mxarray-h.sh)
 
 %reldir%/oct-tex-lexer.ll: %reldir%/oct-tex-lexer.in.ll %reldir%/oct-tex-symbols.in | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
 	$(AWK) 'BEGIN { print "/* DO NOT EDIT. AUTOMATICALLY GENERATED FROM oct-tex-lexer.in.ll and oct-tex-symbols.in. */"; } /^@SYMBOL_RULES@$$/ { count = 0; while (getline < "$(srcdir)/%reldir%/oct-tex-symbols.in") { if ($$0 !~ /^#.*/ && NF == 3) { printf("\"\\\\%s\" { yylval->sym = %d; return SYM; }\n", $$1, count); count++; } } getline } ! /^@SYMBOL_RULES@$$/ { print }' $< > $@-t && \
 	mv $@-t $@
 
 %reldir%/oct-tex-symbols.cc: %reldir%/oct-tex-symbols.in | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
@@ -306,14 +306,20 @@ noinst_LTLIBRARIES += \
   $(Z_CPPFLAGS)
 
 %canon_reldir%_libcorefcn_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
 %canon_reldir%_libcorefcn_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS) $(LLVM_CXXFLAGS)
 
 libinterp_EXTRA_DIST += \
   %reldir%/defaults.in.h \
+  %reldir%/genprops.awk \
   %reldir%/graphics.in.h \
+  %reldir%/mk-errno-list.sh \
+  %reldir%/mk-mxarray-h.in.sh \
   %reldir%/mxarray.in.h \
   %reldir%/oct-errno.in.cc \
   %reldir%/oct-tex-lexer.in.ll \
   %reldir%/oct-tex-parser.in.yy \
   %reldir%/oct-tex-symbols.in
+
+GEN_CONFIG_SHELL += \
+  %reldir%/mk-mxarray-h.sh
diff --git a/libinterp/dldfcn/config-module.awk b/libinterp/dldfcn/config-module.awk
--- a/libinterp/dldfcn/config-module.awk
+++ b/libinterp/dldfcn/config-module.awk
@@ -80,9 +80,13 @@ BEGIN {
             basename, cppflags[i]);
     printf ("%%canon_reldir%%_%s_la_CXXFLAGS = $(libinterp_liboctinterp_la_CXXFLAGS) %s\n",
             basename, cppflags[i]);
     printf ("%%canon_reldir%%_%s_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED_LDFLAG) %s $(OCT_LINK_OPTS) $(WARN_LDFLAGS)\n",
             basename, ldflags[i]);
     printf ("%%canon_reldir%%_%s_la_LIBADD = $(DLD_LIBOCTINTERP_LIBADD) liboctave/liboctave.la %s $(OCT_LINK_DEPS)\n",
             basename, libraries[i]);
   }
+
+  print "";
+  print "$(srcdir)/%reldir%/module.mk: $(srcdir)/%reldir%/config-module.sh $(srcdir)/%reldir%/config-module.awk $(srcdir)/%reldir%/module-files";
+  print "	$(AM_V_GEN)$(SHELL) $(srcdir)/%reldir%/config-module.sh $(srcdir)";
 }
diff --git a/build-aux/mk-builtins.pl b/libinterp/mk-builtins.pl
rename from build-aux/mk-builtins.pl
rename to libinterp/mk-builtins.pl
diff --git a/libinterp/gendoc.pl b/libinterp/mk-doc.pl
rename from libinterp/gendoc.pl
rename to libinterp/mk-doc.pl
--- a/libinterp/gendoc.pl
+++ b/libinterp/mk-doc.pl
@@ -64,17 +64,17 @@ FILE: foreach $fname (@ARGV)
 
       if ($line =~ m#\s*doc:\s+\Q/*\E\s+\Q-*- texinfo -*-\E\s*$#)
       {
         $str = "-*- texinfo -*-\n";
         $reading_docstring = 1;
       }
       else
       {
-        print STDERR "gendoc.pl: undocumented function $func from $fname\n";
+        print STDERR "mk-doc.pl: undocumented function $func from $fname\n";
         push (@docstr, "Undocumented.");
       }
     }
     elsif ($reading_docstring)
     {
       if ($line =~ /^.*\s+\*\/\s*\)\s*$/)
       {
         $line =~ s#\s+\*/\s*\)\s*$##;
diff --git a/libinterp/mk-pkg-add b/libinterp/mk-pkg-add.sh
rename from libinterp/mk-pkg-add
rename to libinterp/mk-pkg-add.sh
diff --git a/build-aux/mk-version-h.in.sh b/libinterp/mk-version-h.in.sh
rename from build-aux/mk-version-h.in.sh
rename to libinterp/mk-version-h.in.sh
diff --git a/libinterp/module.mk b/libinterp/module.mk
--- a/libinterp/module.mk
+++ b/libinterp/module.mk
@@ -78,26 +78,28 @@ LIBINTERP_BUILT_NODISTFILES = \
   %reldir%/version.h \
   %reldir%/builtins.cc
 
 %canon_reldir%_EXTRA_DIST += \
   %reldir%/DOCSTRINGS \
   %reldir%/build-env.in.cc \
   %reldir%/build-env-features.sh \
   %reldir%/deprecated-config.h \
-  %reldir%/gendoc.pl \
-  %reldir%/genprops.awk \
   %reldir%/liboctinterp-build-info.in.cc \
-  %reldir%/mk-errno-list \
-  %reldir%/mk-pkg-add \
-  %reldir%/mkops \
+  %reldir%/mk-builtins.pl \
+  %reldir%/mk-doc.pl \
+  %reldir%/mk-pkg-add.sh \
+  %reldir%/mk-version-h.in.sh \
   %reldir%/op-kw-docs \
   %reldir%/version.in.h \
   $(LIBINTERP_BUILT_DISTFILES)
 
+GEN_CONFIG_SHELL += \
+  %reldir%/mk-version-h.sh
+
 octinclude_HEADERS += \
   %reldir%/builtins.h \
   %reldir%/build-env.h \
   %reldir%/liboctinterp-build-info.h \
   %reldir%/octave.h \
   $(COREFCN_INC) \
   $(LIBINTERP_OPERATORS_INC) \
   $(OCTAVE_VALUE_INC) \
@@ -125,19 +127,16 @@ DIST_SRC += \
 
 include %reldir%/parse-tree/module.mk
 include %reldir%/octave-value/module.mk
 include %reldir%/operators/module.mk
 include %reldir%/template-inst/module.mk
 include %reldir%/corefcn/module.mk
 include %reldir%/dldfcn/module.mk
 
-$(srcdir)/%reldir%/dldfcn/module.mk: $(srcdir)/%reldir%/dldfcn/config-module.sh $(srcdir)/%reldir%/dldfcn/config-module.awk $(srcdir)/%reldir%/dldfcn/module-files
-	$(AM_V_GEN)$(SHELL) $(srcdir)/%reldir%/dldfcn/config-module.sh $(srcdir)
-
 if AMCOND_ENABLE_DYNAMIC_LINKING
   OCT_FILES = $(DLDFCN_LIBS:.la=.oct)
   DLD_LIBOCTINTERP_LIBADD = %reldir%/liboctinterp.la
   LIBINTERP_DLDFCN_LIBADD =
 else
   OCT_FILES =
   DLD_LIBOCTINTERP_LIBADD =
   LIBINTERP_DLDFCN_LIBADD = $(DLDFCN_LIBS)
@@ -236,52 +235,52 @@ nobase_libinterptests_DATA = $(LIBINTERP
 %reldir%/build-env.cc: %reldir%/build-env.in.cc build-aux/subst-config-vals.sh | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)$(call simple-filter-rule,build-aux/subst-config-vals.sh)
 
 %reldir%/build-env-features.cc: config.h %reldir%/build-env-features.sh | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
 	$(SHELL) $(srcdir)/%reldir%/build-env-features.sh $< > $@-t && \
 	mv $@-t $@
 
-%reldir%/version.h: %reldir%/version.in.h build-aux/mk-version-h.sh | %reldir%/$(octave_dirstamp)
-	$(AM_V_GEN)$(call simple-filter-rule,build-aux/mk-version-h.sh)
+%reldir%/version.h: %reldir%/version.in.h %reldir%/mk-version-h.sh | %reldir%/$(octave_dirstamp)
+	$(AM_V_GEN)$(call simple-filter-rule,%reldir%/mk-version-h.sh)
 
 %reldir%/liboctinterp-build-info.cc: %reldir%/liboctinterp-build-info.in.cc HG-ID | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)$(build-info-commands)
 
 if AMCOND_ENABLE_DYNAMIC_LINKING
   mkbuiltins_dld_opt =
 else
   mkbuiltins_dld_opt = --disable-dl
 endif
 
-%reldir%/builtins.cc: $(LIBINTERP_DEFUN_FILES) build-aux/mk-builtins.pl | %reldir%/$(octave_dirstamp)
+%reldir%/builtins.cc: $(LIBINTERP_DEFUN_FILES) %reldir%/mk-builtins.pl | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
-	$(PERL) $(srcdir)/build-aux/mk-builtins.pl --source $(mkbuiltins_dld_opt) "$(srcdir)" -- $(LIBINTERP_DEFUN_FILES) > $@-t && \
+	$(PERL) $(srcdir)/%reldir%/mk-builtins.pl --source $(mkbuiltins_dld_opt) "$(srcdir)" -- $(LIBINTERP_DEFUN_FILES) > $@-t && \
 	mv $@-t $@
 
-%reldir%/builtin-defun-decls.h: $(LIBINTERP_DEFUN_FILES) build-aux/mk-builtins.pl | %reldir%/$(octave_dirstamp)
+%reldir%/builtin-defun-decls.h: $(LIBINTERP_DEFUN_FILES) %reldir%/mk-builtins.pl | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
-	$(PERL) $(srcdir)/build-aux/mk-builtins.pl --header $(mkbuiltins_dld_opt) "$(srcdir)" -- $(LIBINTERP_DEFUN_FILES) > $@-t && \
+	$(PERL) $(srcdir)/%reldir%/mk-builtins.pl --header $(mkbuiltins_dld_opt) "$(srcdir)" -- $(LIBINTERP_DEFUN_FILES) > $@-t && \
 	$(simple_move_if_change_rule)
 
 if AMCOND_ENABLE_DYNAMIC_LINKING
 DLDFCN_PKG_ADD_FILE = %reldir%/dldfcn/PKG_ADD
 
-%reldir%/dldfcn/PKG_ADD: $(DLDFCN_DEFUN_FILES) %reldir%/mk-pkg-add | %reldir%/$(octave_dirstamp)
+%reldir%/dldfcn/PKG_ADD: $(DLDFCN_DEFUN_FILES) %reldir%/mk-pkg-add.sh | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
-	$(SHELL) $(srcdir)/%reldir%/mk-pkg-add "$(srcdir)" $(DLDFCN_DEFUN_FILES) > $@-t && \
+	$(SHELL) $(srcdir)/%reldir%/mk-pkg-add.sh "$(srcdir)" $(DLDFCN_DEFUN_FILES) > $@-t && \
 	mv $@-t $@
 endif
 
 DOCSTRING_FILES += %reldir%/DOCSTRINGS
 
 %reldir%/DOCSTRINGS: $(LIBINTERP_DEFUN_FILES) %reldir%/op-kw-docs | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f %reldir%/DOCSTRINGS-t && \
-	( $(PERL) $(srcdir)/%reldir%/gendoc.pl "$(srcdir)" $(LIBINTERP_DEFUN_FILES); $(SED) -ne '/^\x1d/,$$p' $(srcdir)/%reldir%/op-kw-docs ) > %reldir%/DOCSTRINGS-t && \
+	( $(PERL) $(srcdir)/%reldir%/mk-doc.pl "$(srcdir)" $(LIBINTERP_DEFUN_FILES); $(SED) -ne '/^\x1d/,$$p' $(srcdir)/%reldir%/op-kw-docs ) > %reldir%/DOCSTRINGS-t && \
 	$(call move_if_change_rule,%reldir%/DOCSTRINGS-t,$@)
 
 OCTAVE_INTERPRETER_TARGETS += \
   $(OCT_FILES) \
   $(DLDFCN_PKG_ADD_FILE) \
   $(LIBINTERP_TST_FILES)
 
 DIRSTAMP_FILES += %reldir%/$(octave_dirstamp)
diff --git a/libinterp/mkops b/libinterp/operators/mk-ops.sh
rename from libinterp/mkops
rename to libinterp/operators/mk-ops.sh
--- a/libinterp/mkops
+++ b/libinterp/operators/mk-ops.sh
@@ -16,17 +16,17 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 SED=${SED:-sed}
 
 cat << \EOF
-// DO NOT EDIT!  Generated automatically by mkops.
+// DO NOT EDIT!  Generated automatically by mk-ops.sh.
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 extern void install_base_type_conversions (void);
 
 EOF
diff --git a/libinterp/operators/module.mk b/libinterp/operators/module.mk
--- a/libinterp/operators/module.mk
+++ b/libinterp/operators/module.mk
@@ -129,13 +129,16 @@ LIBINTERP_OPERATORS_INC = \
 
 ## These look like included header files to Autotools build process
 NOINSTALL_LIBINTERP_OPERATORS_INC = \
   %reldir%/op-dm-template.cc \
   %reldir%/op-dms-template.cc \
   %reldir%/op-int.h \
   %reldir%/op-pm-template.cc
 
+libinterp_EXTRA_DIST += \
+  %reldir%/mk-ops.sh
+
 ## Special rules for sources which must be built before rest of compilation.
-%reldir%/ops.cc: $(LIBINTERP_OPERATORS_SRC) libinterp/mkops
+%reldir%/ops.cc: $(LIBINTERP_OPERATORS_SRC) %reldir%/mk-ops.sh
 	$(AM_V_GEN)rm -f $@-t $@ && \
-	$(SHELL) $(srcdir)/libinterp/mkops $(LIBINTERP_OPERATORS_SRC) > $@-t && \
+	$(SHELL) $(srcdir)/%reldir%/mk-ops.sh $(LIBINTERP_OPERATORS_SRC) > $@-t && \
 	mv $@-t $@
diff --git a/build-aux/mk-f77-def.in.sh b/liboctave/external/mk-f77-def.in.sh
rename from build-aux/mk-f77-def.in.sh
rename to liboctave/external/mk-f77-def.in.sh
diff --git a/liboctave/external/module.mk b/liboctave/external/module.mk
--- a/liboctave/external/module.mk
+++ b/liboctave/external/module.mk
@@ -18,19 +18,19 @@ include %reldir%/fftpack/module.mk
 include %reldir%/lapack-xtra/module.mk
 include %reldir%/odepack/module.mk
 include %reldir%/ordered-qz/module.mk
 include %reldir%/quadpack/module.mk
 include %reldir%/ranlib/module.mk
 include %reldir%/slatec-err/module.mk
 include %reldir%/slatec-fn/module.mk
 
-%reldir%/external.def: $(%canon_reldir%_libexternal_la_SOURCES) build-aux/mk-f77-def.sh
+%reldir%/external.def: $(%canon_reldir%_libexternal_la_SOURCES) %reldir%/mk-f77-def.sh
 	$(AM_V_GEN)rm -f $@-t $@ && \
-	$(SHELL) build-aux/mk-f77-def.sh $(srcdir) $(%canon_reldir%_libexternal_la_SOURCES) > $@-t && \
+	$(SHELL) %reldir%/mk-f77-def.sh $(srcdir) $(%canon_reldir%_libexternal_la_SOURCES) > $@-t && \
 	mv $@-t $@
 
 liboctave_CLEANFILES += \
   %reldir%/external.def \
   %reldir%/ranlib/ranlib.def \
   $(nodist_%canon_reldir%_libexternal_la_SOURCES)
 
 noinst_LTLIBRARIES += %reldir%/libexternal.la
@@ -39,8 +39,14 @@ noinst_LTLIBRARIES += %reldir%/libextern
 
 %canon_reldir%_libexternal_la_CPPFLAGS = $(liboctave_liboctave_la_CPPFLAGS)
 
 %canon_reldir%_libexternal_la_CFLAGS = $(liboctave_liboctave_la_CFLAGS)
 
 %canon_reldir%_libexternal_la_CXXFLAGS = $(liboctave_liboctave_la_CXXFLAGS)
 
 liboctave_liboctave_la_LIBADD += %reldir%/libexternal.la
+
+liboctave_EXTRA_DIST += \
+  %reldir%/mk-f77-def.in.sh
+
+GEN_CONFIG_SHELL += \
+  %reldir%/mk-f77-def.sh
diff --git a/liboctave/external/ranlib/module.mk b/liboctave/external/ranlib/module.mk
--- a/liboctave/external/ranlib/module.mk
+++ b/liboctave/external/ranlib/module.mk
@@ -38,19 +38,19 @@ RANLIB_SRC = \
 noinst_LTLIBRARIES += %reldir%/libranlib.la
 
 %canon_reldir%_libranlib_la_SOURCES = $(RANLIB_SRC)
 
 %canon_reldir%_libranlib_la_DEPENDENCIES = %reldir%/ranlib.def
 
 ## Special rules for files which must be built before compilation
 ## ranlib directory may not exist in VPATH build; create it if necessary.
-%reldir%/ranlib.def: $(RANLIB_SRC) build-aux/mk-f77-def.sh | %reldir%/$(octave_dirstamp)
+%reldir%/ranlib.def: $(RANLIB_SRC) %reldir%/../mk-f77-def.sh | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
-	$(SHELL) build-aux/mk-f77-def.sh $(srcdir) $(RANLIB_SRC) > $@-t && \
+	$(SHELL) %reldir%/../mk-f77-def.sh $(srcdir) $(RANLIB_SRC) > $@-t && \
 	mv $@-t $@
 
 liboctave_liboctave_la_LIBADD += %reldir%/libranlib.la
 
 liboctave_EXTRA_DIST += \
   %reldir%/Basegen.doc \
   %reldir%/HOWTOGET \
   %reldir%/README \
diff --git a/scripts/mkdoc.pl b/scripts/mk-doc.pl
rename from scripts/mkdoc.pl
rename to scripts/mk-doc.pl
diff --git a/scripts/mk-pkg-add b/scripts/mk-pkg-add.sh
rename from scripts/mk-pkg-add
rename to scripts/mk-pkg-add.sh
diff --git a/scripts/module.mk b/scripts/module.mk
--- a/scripts/module.mk
+++ b/scripts/module.mk
@@ -86,36 +86,36 @@ ALL_LOCAL_TARGETS += \
 
 OCTAVE_INTERPRETER_TARGETS += \
   $(GEN_FCN_FILES) \
   $(PKG_ADD_FILES)
 
 FCN_FILES_WITH_TESTS = $(shell $(SHELL) $(srcdir)/build-aux/find-files-with-tests.sh "$(srcdir)" $(FCN_FILES) $(GEN_FCN_FILES_IN))
 
 define PKG_ADD_FILE_TEMPLATE
-$(1)/PKG_ADD: $$($(2)_FCN_FILES) $$($(2)_GEN_FCN_FILES) $(1)/$(octave_dirstamp) %reldir%/mk-pkg-add
+$(1)/PKG_ADD: $$($(2)_FCN_FILES) $$($(2)_GEN_FCN_FILES) $(1)/$(octave_dirstamp) %reldir%/mk-pkg-add.sh
 	$$(AM_V_GEN)rm -f $$@-t $$@ && \
-	$$(SHELL) $$(srcdir)/%reldir%/mk-pkg-add $(srcdir) $$($(2)_FCN_FILES) -- $$($(2)_GEN_FCN_FILES) > $$@-t && \
+	$$(SHELL) $$(srcdir)/%reldir%/mk-pkg-add.sh $(srcdir) $$($(2)_FCN_FILES) -- $$($(2)_GEN_FCN_FILES) > $$@-t && \
 	mv $$@-t $$@
 endef
 
 $(foreach d, $(FCN_FILE_DIRS), $(eval $(call PKG_ADD_FILE_TEMPLATE, $(d),$(subst /,_,$(subst -,_,$(d))))))
 
 define GEN_FCN_FILES_TEMPLATE
 $(1): $(1:.m=.in.m) build-aux/subst-config-vals.sh $(dir $(1))$(octave_dirstamp)
 	$$(AM_V_GEN)$$(call simple-filter-rule,build-aux/subst-config-vals.sh)
 endef
 
 $(foreach f, $(GEN_FCN_FILES), $(eval $(call GEN_FCN_FILES_TEMPLATE, $(f))))
 
 DOCSTRING_FILES += %reldir%/DOCSTRINGS
 
 %reldir%/DOCSTRINGS: $(FCN_FILES) $(GEN_FCN_FILES_IN) | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f %reldir%/DOCSTRINGS-t && \
-	$(PERL) $(srcdir)/%reldir%/mkdoc.pl "$(srcdir)" $(FCN_FILES) $(GEN_FCN_FILES_IN) > %reldir%/DOCSTRINGS-t && \
+	$(PERL) $(srcdir)/%reldir%/mk-doc.pl "$(srcdir)" $(FCN_FILES) $(GEN_FCN_FILES_IN) > %reldir%/DOCSTRINGS-t && \
 	$(call move_if_change_rule,%reldir%/DOCSTRINGS-t,$@)
 
 DIRSTAMP_FILES += %reldir%/$(octave_dirstamp)
 
 check-m-sources:
 	@echo "checking whether files in source tree are listed in module.mk files..."; \
 	for f in $$(find $(srcdir)/scripts -name '*.m'); do \
 	  found=false; \
@@ -181,18 +181,18 @@ scripts-dist-hook:
 	@echo "Packaging distribution requires Java." ; exit 1;
 endif
 
 %canon_reldir%_EXTRA_DIST += \
   $(SCRIPTS_IMAGES) \
   $(FCN_FILES) \
   $(GEN_FCN_FILES_IN) \
   %reldir%/DOCSTRINGS \
-  %reldir%/mkdoc.pl \
-  %reldir%/mk-pkg-add
+  %reldir%/mk-doc.pl \
+  %reldir%/mk-pkg-add.sh
 
 EXTRA_DIST += $(%canon_reldir%_EXTRA_DIST)
 
 %canon_reldir%_CLEANFILES += \
   $(GEN_FCN_FILES) \
   $(PKG_ADD_FILES)
 
 %canon_reldir%_DISTCLEANFILES += \
